packages/stream_chat_flutter/lib/src/mentions_overlay.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:flutter/material.dart';
+import 'package:stream_chat_flutter/stream_chat_flutter.dart';
+
+/// Overlay for displaying users that can be mentioned
+class MentionsOverlay extends StatelessWidget {
+  /// Constructor for creating a [MentionsOverlay]
+  const MentionsOverlay({
+    required this.query,
+    required this.onMentionResult,
+    required this.size,
+    required this.channel,
+    this.mentionsTileBuilder,
+    Key? key,
+  }) : super(key: key);
+
+  /// The size of the overlay
+  final Size size;
+
+  /// Query for searching users
+  final String query;
+
+  /// The channel to search for users
+  final Channel channel;
+
+  /// Callback called when a user is selected
+  final ValueChanged<User?> onMentionResult;
+
+  /// Customize the tile for the mentions overlay
+  final MentionTileBuilder? mentionsTileBuilder;
+
+  @override
+  Widget build(BuildContext context) {
+    final _streamChatTheme = StreamChatTheme.of(context);
+
+    Future<List<Member>>? queryMembers;
+
+    if (query.isNotEmpty) {
+      queryMembers = channel
```

Comment:
We can't do this in the build. This will be triggered unnecessarily and also results in the current flickering. This should be a StateFullWidget that only calls the future when the query changes, use didUpdateWidget

Reply:
yes, we're working on this